SalesforceCommerceCloud/pwa-kit
# @W-16564111 Shopper Context in PWA Kit

# Description



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- add ability to set shopper context via URL search params to template-retail-react-app

# How to Test-Drive This PR

- Checkout this branch `git checkout W-16564111-shopper-context-in-pwa-kit`
- Start template-retail-react-app `cd packages/template-retail-react-app && npm start`
    - If the token endpoint is returning a 400 error, you may need to clear the site data by opening the developer tools, going to the *Application* tab and clicking *Clear Site Data*
- Navigate around the site without any query params and verify the shopper-context API is not called
- Navigate to http://localhost:3000/global/en-GB/category/womens-clothing-tops 
- Set `sourceCode=instagram` in the Shopper Context by navigating to http://localhost:3000/global/en-GB/category/womens-clothing-tops/?sourceCode=instagram
  - Verify the prices have been updated and each product contains the promotion `Exclusive for Instagram users`
- In constant.js, add fields to `SHOPPER_CONTEXT_SEARCH_PARAMS.customQualifiers` and `SHOPPER_CONTEXT_SEARCH_PARAMS.assignmentQualifiers`
    ```
    export const SHOPPER_CONTEXT_SEARCH_PARAMS = {
        ...
        customQualifiers: {deviceType: {paramName: "deviceType"}},
        assignmentQualifiers: {store: {paramName: "store"}}
    }
    ```
- Verify all shopper context can be added via search params: http://localhost:3000/us/en-US/?sourceCode=instagram&effectiveDateTime=2020-12-20T00:00:00Z&deviceType=abc&store=boston&customerGroupIds=BigSpenders&customerGroupIds=LowSpenders&clientIp=13.108.0.0&countryCode=CA&city=toronto&latitude=37.7749&longitude=37.7749&metroCode=AB&postalCode=A3B2C5&region=soemwhere&regionCode=ZZ
- In constants.js, update the `paramName` of `sourceCode` in `SHOPPER_CONTEXT_SEARCH_PARAMS` to `promo`
    ```
    export const SHOPPER_CONTEXT_SEARCH_PARAMS = {
        shopperContext: {
            sourceCode: {paramName: 'promo'},
        ...
    ```
- Verify navigating to http://localhost:3000/global/en-GB/category/womens-clothing-tops/?promo=instagram sets the shopper context source code to `instagram`

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
